src/Aspire.Cli/Utils/ConfigurationHelper.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Aspire.Cli.Utils;
+
+internal static class ConfigurationHelper
+{
+    internal static void RegisterSettingsFiles(IConfigurationBuilder configuration, DirectoryInfo workingDirectory, FileInfo globalSettingsFile)
+    {
+        var currentDirectory = workingDirectory;
+
+        // Find the nearest local settings file
+        FileInfo? localSettingsFile = null;
+
+        while (currentDirectory is not null)
+        {
+            var settingsFilePath = Path.Combine(currentDirectory.FullName, ".aspire", "settings.json");
+
+            if (File.Exists(settingsFilePath))
+            {
+                localSettingsFile = new FileInfo(settingsFilePath);
+                break;
+            }
+
+            currentDirectory = currentDirectory.Parent;
+        }
+
+        // Add local settings first (if found)
+        if (localSettingsFile is not null)
+        {
+            configuration.AddJsonFile(localSettingsFile.FullName, optional: true);
+        }
+
+        // Then add global settings file (if it exists) - this will override local settings
```

Comment:
This is backwards. Local always overrides global.

Reply:
I am glad you brought this up. This is one of the reasons I wasn't super excited about doing the global stuff up front because I am not sure local overrides global holds true all the time. For example container runtime?

A team might decide that they want to use podman and so they set their default in their repo local to use podman. But someone wants to use Docker Desktop for some reason - how do they override it?

There are probably other examples.

Instinctively we think local overrides global, but when it comes to a repository that is cloned down what is considered "local"?